Documentation/contributing/testing/scalability.rst

```diff
@@ -0,0 +1,294 @@
+  and measures the time it takes to resolve FQDNs from client point of view.
+
+* ClusterMesh scale test - ``/scale-clustermesh`` `Workflow <CLUSTERMESH_WORKFLOW_>`_ leverages
+  a `mock Clustermesh control-plane <CLUSTERMESH_MOCK_>`_ that simulates large deployments of ClusterMesh.
+
+Test based on EKS:
+
+* Egress Gateway scale test - ``/scale-egw``. `Workflow <EGW_WORKFLOW_>`_ tests Egress Gateway on a small cluster,
+  but with synthetically created Endpoints and Nodes to simulate large cluster.
+
+Whenever developing new tests, you should consider if you want to add a test to already existing workflow,
+create a new one or maybe extend the existing test.
+If you are unsure, you can always ask in the ``#sig-scalabilty`` `Slack channel <SLACK_CHANNEL_>`_.
+For example, if you want to run test on a large cluster,
+you might consider adding it as a separate test scenario to already existing 100 nodes scale test
+to reduce cost of CI, as spinning up a new cluster and tearing it down is quite a long process.
+For some use-cases, it might be better to simulate large cluster and execute test on a small cluster,
+like in the case of Egress Gateway scale test or ClusterMesh scale test.
+
+Running CL2 tests locally
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each CL2 test should be designed in a way that it scales with number of nodes.
+This allows for running specific test case scenario in a local environment, to validate the test case.
+For example, let's run network policy scale test in local kind cluster.
+First, setup a kind cluster with Cilium as documented in :ref:`dev_env`.
+Build ClusterLoader2 binary from `perf-tests repository <CL2_>`_.
+Then you can run:
+
+.. code-block:: bash
+
+    export CL2_PROMETHEUS_PVC_ENABLED=false
+    export CL2_PROMETHEUS_SCRAPE_CILIUM_OPERATOR=true
+    export CL2_PROMETHEUS_SCRAPE_CILIUM_AGENT=true
+    export CL2_PROMETHEUS_SCRAPE_CILIUM_AGENT_INTERVAL=5s
+
+    ./clusterloader \
+    -v=2 \
+    --testconfig=.github/actions/cl2-modules/netpol/config.yaml \
+    --provider=kind \
+    --enable-prometheus-server \
+    --nodes=1 \
+    --report-dir=./report \
+    --prometheus-scrape-kube-proxy=false \
+    --prometheus-apiserver-scrape-port=6443 \
+    --kubeconfig=$HOME/.kube/config
+
+
+Some additional options worth mentioning are:
+
+* ``--tear-down-prometheus-server=false`` - Leaves Prometheus and Grafana running after the test finishes, this helps with speeding up the test run
+  when running multiple tests in a row, but also for exploring the metrics in Grafana.
+* ``--experimental-prometheus-snapshot-to-report-dir=true`` - Creates a snapshot of the Prometheus data and saves it to the report directory
+
+By setting ``deleteAutomanagedNamespaces: false`` in the test config, you can also leave
+the test namespaces after the test finishes. This is especially useful for checking if your test
+created the expected resources.
+
+At the end of output, test should end successfully with:
+
+.. code-block:: bash
```

Comment:
This is no bash code. Note the double-colon at the end of the line, to open the preformatted block.

```suggestion
At the end of output, the test should end successfully with::
```